runtime/j9vm/javanextvmi.cpp

```diff
@@ -609,7 +622,7 @@ JVM_VirtualThreadHideFrames(
 
 	if (!hide) {
 		Assert_SC_true(hiddenFrames);
-		exitVThreadTransitionCritical(currentThread, vThreadObj);
+		exitVThreadTransitionCritical(currentThread, (jobject)&currentThread->threadObject);
```

Comment:
actually this should be fine, since the jobject passed in is the address of `currentThread->threadObject` even if the `threadObject` is moved by GC, the new address should be written to the same address by GC